addon/components/impagination-collection.js

```diff
@@ -44,11 +38,15 @@ export default Ember.Component.extend({
     this.set('datasetState', this.get('dataset.state'));
   }),
 
+  safeSet: function(key, value) {
+    if (!this.isDestroyed) { this.set(key, value); }
+  },
+
   didInitAttrs() {
     this._super.apply(this, arguments);
 
     this.setInitialState();
-    this.get('dataset').setReadOffset(this.get('initialReadOffset') || 0);
+    this.get('dataset').setReadOffset(this.get('initial-read-offset') || 0);
```

Comment:
Do you want this to be able to be changed at any time? Because right now we only use this on didInitAttrs hooks. Possibly utilize didReceiveAttrs?